pkg/core/multi_search.go

```diff
@@ -30,31 +30,58 @@ import (
 	"github.com/zinclabs/zinc/pkg/uquery/timerange"
 )
 
+// isMatchIndex("abc", "a")  false
+// isMatchIndex("abc", "a*") true
+// isMatchIndex("abc", "*bc") true
+// isMatchIndex("abc", "bc") false
+// isMatchIndex("abc", "abc") true
+func isMatchIndex(zincIndexName, indexName string) bool {
+	// the old code: indexName == "" || (indexName != "" && strings.HasPrefix(zincIndexName, indexName[:len(indexName)-1]))
+
+	if indexName == "" {
+		return true
+	}
+	name := indexName
+	// eg.: *-test
+	if strings.HasPrefix(indexName, "*") {
+		name = indexName[1:len(indexName)] // eg.: *-test -> -test
+		return strings.HasSuffix(zincIndexName, name)
+	}
+
+	// eg.: test-*
+	if strings.HasSuffix(indexName, "*") {
+		name = indexName[:len(indexName)-1] // eg.: test-* -> test-
+		return strings.HasPrefix(zincIndexName, name)
+	}
+
+	return zincIndexName == indexName
+}
+
 func MultiSearch(indexNames []string, query *meta.ZincQuery) (*meta.SearchResponse, error) {
 	var mappings *meta.Mappings
 	var analyzers map[string]*analysis.Analyzer
 	var readers []*bluge.Reader
 	var shardNum int
-	indexMap := make(map[string]struct{})
 
 	timeMin, timeMax := timerange.Query(query.Query)
 	for _, index := range ZINC_INDEX_LIST.List() {
+		isMatched := false
 		for _, indexName := range indexNames {
-			if _, ok := indexMap[index.Name]; ok {
-				continue
+			isMatched = isMatchIndex(index.Name, indexName)
+			if isMatched {
+				break
+			}
+		}
+		if isMatched {
```

Comment:
We can set `if !isMatched` then break, then we will reduce one nest.

Reply:
If set if !isMatched then break, The following ZincIndexNames haven't check match yet，I think we need check left in “ MultiSearch method”。What about you?

Reply:
Sorry, i made a mistake, i mean we should if `!isMatched then `continue`, then we will reduce one nest, the code format will be better.

Reply:
Yes ，its better